.circleci/config.yml

```diff
@@ -26,6 +26,7 @@ jobs:
       - run: pip install -r requirements.txt
       - run: black --check .
       - run: mypy algosdk
+      - run: python setup.py check --strict
```

Comment:
@algochoi Surprisingly, the check does _not_ catch warnings from normalized versions.  Otherwise, we'd expect the following build to fail:  https://app.circleci.com/pipelines/github/algorand/py-algorand-sdk/1174/workflows/147f202e-e47f-4998-b5e8-4437e5d8a0cc/jobs/6005.

In my brief search, I didn't find an explanation for _why_ the check completes _with_ the warning present.

Have I missed an obvious solution here?  If not, are you opposed to extending the check to confirm exit code = 0 _and_ no warnings in output?

Reply:
Yes, your observation is correct. I'm not opposed to adding that check, maybe something like the following to output and check the results of the command:
```
python setup.py check -s 2>&1 | tee /dev/tty | grep --quiet "warning"; test $? -eq 1
```

Reply:
@algochoi As of https://github.com/algorand/py-algorand-sdk/pull/427/commits/2fa369559fb601533c5d9b6e5f5ad57ec6bafddc, I think the check catches the normalizing error, but does _not_ catch errors emitted by `setup.py`.  If you agree with the observation, can you review the following proposal?

Here's a proposal based on local testing:

```
python setup.py check -s 2>&1 | tee /dev/tty | (! grep -qEi 'error|warning')
```

I tested against 2 commits:
* `e90337dcf0dfc9d9b63c5a786c22228058379a38` - `master` as of writing containing _only_ normalization warning (`setup.py check` exit code = 0).
* `a4114577c41ba0abcd5f4fad50419c6abc0ba6e9` - Arbitrary commit containing URL metadata error + normalization warning.

Reply:
The exit code looks like it is returned correctly (both the current check and your proposal), but it's not output to the console. I proposed the following in [6fd99ca](https://github.com/algorand/py-algorand-sdk/pull/427/commits/6fd99caa78dbd19e8cb1a446bf5a500123971aa3):

```
- run: |
          python setup.py check -s
          ! python setup.py check -s 2>&1 | grep --quiet --ignore-case "warning"
```

The first line should fail for any errors anyway and output all errors and warnings. The second line is there to fail if there are any warnings.